pydal/objects.py

```diff
@@ -2386,6 +2386,12 @@ def __str__(self):
         self.export_to_csv_file(s)
         return s.getvalue()
 
+    def __eq__(self, other):
+        if len(self.records) != len(other.records):
+            return False
+        l = map(lambda x: x[0] == x[1], zip(self.records, other.records))
```

Comment:
given the recent "discoveries" of iterRows and the fact that DAL can be less memory hungry if coded efficiently....you took the time to code an __eq__: you should just break at the first difference found, without looping through the entire resultset and then checking if all the checks are True or False...think 1M:1M Rows to compare......